src/main/java/org/folio/service/protection/ProtectionServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package org.folio.service.protection;
+
+import static org.folio.exception.ErrorCodes.ORGANIZATION_UNITS_NOT_FOUND;
+import static org.folio.exception.ErrorCodes.USER_HAS_NO_PERMISSIONS;
+import static org.folio.rest.RestVerticle.OKAPI_USERID_HEADER;
+
+import io.vertx.core.Context;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.folio.HttpStatus;
+import org.folio.exception.HttpException;
+import org.folio.rest.acq.model.AcquisitionsUnit;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.service.BaseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+
+@Service
+public class ProtectionServiceImpl extends BaseService implements ProtectionService {
+  private AcquisitionsUnitsService acquisitionsUnitsService;
+  private final List<AcquisitionsUnit> fetchedUnits = new ArrayList<>();
+
+  @Autowired
+  public void setAcquisitionsUnitsService(AcquisitionsUnitsService acquisitionsUnitsService) {
+    this.acquisitionsUnitsService = acquisitionsUnitsService;
+  }
+
+  @Override
+  public CompletableFuture<Void> isOperationRestricted(List<String> unitIds, Set<ProtectedOperationType> operations, String lang, Context context, Map<String, String> headers) {
```

Comment:
method name and return type confuses a bit. Having "isOperationRestricted" means that method will return Boolean value, but returns Void. Please use more meaningful naming or change returning type

Reply:
renamed